### ISSUE TITLE
[data grid] Keyboard navigation (Page Up, Page Down, Space) not working

### ISSUE THREAD

**Issue by ramyab03** [mui/mui-x#4749]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

When we set custom height to DataGrid row (rowHeight={100}) - keyboard navigation with Pageup, Page Down, Space are not working as expected. When we try Page Down - after a couple of times the  scrolling stops (gets stuck). Same issue with Page Up as well.

### Expected behavior 🤔

When we use page down, space - we should be able to scroll to the next set of results in the Grid without stopping/getting stuck.

### Steps to reproduce 🕹

Steps:

1. Open https://codesandbox.io/s/eqtdrl?file=/demo.tsx
2. Select any cell from the first row
3. Click the Space key a couple of times

### Context 🔦

This will affect any keyboard users who are accessing the application.

### Your environment 🌎

Browser - Chrome

![image](https://user-images.githubusercontent.com/44844109/166471180-def26016-495c-4e63-bd22-f5ee1fabebd0.png)

**Comment by DanailH** [comment]
Hi, @ramyab03 thanks for raising this. I tested it and it is a valid problem.

**Comment by ramyab03** [comment]
@DanailH - Please let us know if you have a workaround for this issue. 

For future can you also let us know if this problem will be addressed in the upcoming releases.

Thank You

**Comment by alexfauquette** [comment]
This problem also occurs when using `autoHeight`

https://codesandbox.io/s/autoheightgrid-demo-mui-x-forked-44p5z6?file=/demo.tsx
To reproduce, click on a cell to set focus and press <kbd>Space</kbd>.

The focus will move to the last row but will not scroll so the cell is not rendered

**Comment by oliviertassinari** [comment]
I could reproduce with the default props, https://codesandbox.io/s/denseheightgrid-material-demo-forked-ucqep8?file=/demo.tsx

```jsx
import * as React from "react";
import { DataGrid } from "@mui/x-data-grid";
import { useDemoData } from "@mui/x-data-grid-generator";

export default function DenseHeightGrid() {
  const { data } = useDemoData({
    dataSet: "Commodity",
    rowLength: 100,
    maxColumns: 6
  });

  return (
    <div style={{ height: 400, width: "100%" }}>
      <DataGrid {...data} />
    </div>
  );
}
```

https://user-images.githubusercontent.com/3165635/180210167-7ca35718-6951-4880-abfa-92072a5b7f7c.mov

**Comment by michelengelen** [comment]
Thanks for the report and the repro. The underlying problem was that virtualization unmounted the focused cell, so keyboard navigation lost its anchor and got stuck. This was fixed in v6: #7357 keeps the focused cell in the DOM (released in v6.10.2), and #9203 fixed the focus update when PageUp/PageDown crosses pages (v6.10.1). PageUp, PageDown, and Space handling in the current virtualizer is covered by unit tests, including custom row heights.

Closing as completed. If any of these keys still stick on the latest version, please open a new issue with a sandbox.
